File: WINGS_AI_Bootcamp/scala_Ymeg/armstrong.scala
import scala.io.StdIn.readInt

def countDigits(n: Int): Int =
  n.toString.length

def powerSum(n: Int, digits: Int): Int =
  n.toString.map(_.asDigit).map(d => Math.pow(d, digits)).sum.toInt

def isArmstrong(n: Int): Boolean =
  val digits = countDigits(n)
  powerSum(n, digits) == n

def generateArmstrong(limit: Int): List[Int] =
  (1 to limit).filter(isArmstrong).toList

@main def run(): Unit =
  val n = readInt()
  println(generateArmstrong(n))

File: WINGS_AI_Bootcamp/Elixir/ArmStrong.ex
defmodule Armstrong do
  def is_armstrong(n) do
    digits = Integer.digits(n)
    power = length(digits)
    sum = compute_sum(digits, power)
    sum == n
  end

  defp compute_sum([], _), do: 0
  defp compute_sum([h | t], p), do: :math.pow(h, p) + compute_sum(t, p)
end

IO.inspect(Armstrong.is_armstrong(153))  # true

